### ISSUE TITLE
Code highlighting support

### ISSUE THREAD

**Issue by mre** [cobalt-org/cobalt.rs#49]
Hi,

I'm currently moving my homepage from jekyll to cobalt.rs and I love it! Thanks for this tool. :smiley: 

One thing that I'm missing is code highlighting. While I could use external javascript highlighters it would be nice to have a pure html/css solution.

Jekyll supports code snippet highlighting as a filter: http://jekyllrb.com/docs/templates/#code-snippet-highlighting. Maybe something like this can also be possible in cobalt.rs? Thinking a bit outside the box it would be nice to have a simple plugin system for things like that.
I've actually written a super simple wrapper around pygments that I try to use for syntax highlighting.
(See https://github.com/mre/pygments-rs)

Do you think something like this might make sense?

**Comment by johannhof** [comment]
Absolutely.

This sounds like a good use case for writing a [liquid](https://github.com/cobalt-org/liquid-rust) plugin. It would work exactly like the Jekyll version:

```
{% highlight ruby %}
def foo
  puts 'foo'
end
{% endhighlight %}
```

Without having looked it up, I'm pretty sure that's also just a liquid filter in Jekyll.

Unfortunately, writing tags for liquid-rust is not very well documented right now. I'd love to help you though! Take a look at other tag implementations like https://github.com/cobalt-org/liquid-rust/blob/master/src/tags/for_block.rs and if you have any questions, you can also join our [chats](https://github.com/cobalt-org/cobalt.rs/blob/master/CONTRIBUTING.md)

Long term, a plugin-like system for Cobalt itself also sounds awesome, especially as Cobalt development happens more on a "what-do-contributors-need-for-their-website" basis right now.

**Comment by johannhof** [comment]
If you don't have the time/energy to spend on writing a highlight plugin for liquid-rust right now, I can also put it on my personal list, but it might take a bit longer then :sweat_smile:

**Comment by mre** [comment]
Okay cool. Thanks for the fast answer. Really appreciate the links.
I'll give it a try if I find the time. But to be honest I'm unsure when that will be. Maybe we can keep this on your personal list as well ;-)

**Comment by johannhof** [comment]
Ok, I'm actually pretty interested in this as well, so I'll try to squeeze it in :squirrel:

**Comment by douglascorrea** [comment]
One question here about plugins: they should be focused in rust-liquid or should cobalt.rs also support plugin? I see like paginatiom or multi language functionality will work great as cobalt.rs plugin

**Comment by johannhof** [comment]
Well right now it's easiest to create liquid plugins that we can just put into Cobalt at compile time. That would be code highlighting, for example. I'd also like to write a good inheritance/mixin plugin for liquid.

Plugins for Cobalt itself are a bit trickier, since it's statically compiled and we probably want people to write their plugins in Rust as well. I love the idea though, we should put it on a long-term plan.

**Comment by johannhof** [comment]
Experimental support landed in #165 🎉
